hotshot-state-prover/src/bin/state-prover.rs

```diff
@@ -170,44 +170,45 @@ async fn main() {
         max_gas_price,
     };
 
-    // TODO(Chengyu): turn on the v3 prover after the sequencer update
     // validate that the light client contract is a proxy, panics otherwise
     config.validate_light_client_contract().await.unwrap();
-    let is_legacy = match sol_types::LightClient::new(args.light_client_address, &l1_provider)
-        .getVersion()
-        .call()
-        .await
-    {
-        Ok(version) => version.majorVersion == 1,
-        Err(err) => {
-            tracing::error!("Error checking the contract version: {err}");
-            return;
-        },
-    };
-    tracing::info!(
-        "Detected contract version {}",
-        if is_legacy { "v1" } else { "v2" }
-    );
+    let contract_version =
+        match sol_types::LightClient::new(args.light_client_address, &l1_provider)
+            .getVersion()
+            .call()
+            .await
+        {
+            Ok(version) => version.majorVersion,
+            Err(err) => {
+                tracing::error!("Error checking the contract version: {err}");
+                return;
+            },
+        };
+    tracing::info!("Detected contract version v{contract_version}");
 
     // This bind version doesn't represent anything now, but it's required by the service trait
     let bind_version = StaticVersion::<0, 1> {};
 
     if args.daemon {
         // Launching the prover service daemon
-        let result = if is_legacy {
+        let result = if contract_version == 1 {
```

Comment:
let's write in `match case`, it's clearer.

same for the next nested `if else else`